proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,254 @@
+
+# Test Validation Requirements
+The following statements must be true and validated for this work to be
+considered completed.
+- All new test cases pass when run locally against a WARP device
+- All new test cases are confirmed to be lit up and available in the HLK when
+  the target device reports support
+- All new tests/test cases are added to the HLK playlist
+
+# Requirements
+- Greg's long vector changes:
+  https://github.com/microsoft/hlsl-specs/blob/main/proposals/0026-hlsl-long-vector-type.md#allowed-elementwise-vector-intrinsics
+- WARP long vector support (Jesse). Needs Greg's work, or a private WIP
+   branch of Greg's work. ETA of ~1 week to implement.
+
+# Notes
+- The 'ExecTests' (ExecutionTest.cpp) are used to generate the HLK tests
+   (DxilConf tests)
+- Private test binaries/collateral will be shared with IHVs for validation
+   purposes. This will enable IHVs to verify long vector functionality without
+   waiting for an OS/HLK release.
+
+# Open Questions
+- We speculate that the combination of test cases for HLSL operators and HLSL
+intrinsics which map to DXIL ops should get us all, or most of the coverage
+required for the 'emulated' intrinsics. But we should audit that.
+
+# HLSL Operators
+These operators should generate LLVM native ops which use vectors.
+
+Operator table from [Microsoft HLSL Operators](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-operators)
+| Operator Name | Operator | Notes |
+|-----------|--------------|----------|
+Additive and Multiplicative Operators | +, -, *, /, % |
+Array Operator | [i] |
+Assignment Operators | =, +=, -=, *=, /=, %= |
+Binary Casts | asfloat(), asint(), asuint() |
+Bitwise Operators | ~, <<, >>, &, \|, ^, <<=, >>=, &=, \|=, ^= | Only valid on int and uint vectors
+Boolean Math Operators | & &, ||, ?: |
+Cast Operator | (type) |
+Comma Operator | , |
+Comparison Operators| <, >, ==, !=, <=, >= |
+Prefix or Postfix Operators| ++, -- |
+Structure Operator | . |
+Unary Operators | !, -, + |
+
+# Mappings of HLSL intrinsics to DXIL opcodes or LLVM native operations
+
+## Trigonometry
+
+| Intrinsic | DXIL OPCode | Notes |
+|-----------|--------------|----------|
+| acos      | DXIL::OpCode::Acos |  |
+| asin      | DXIL::OpCode::Asin |  |
+| atan      | DXIL::OpCode::Atan |  |
+| atan2     | Emulated |            |
+| cos       | DXIL::OpCode::Cos |   |
+| cosh      | DXIL::OpCode::Hcos |  |
+| degrees   | Emulated |            |
+| radians   | Emulated |            |
+| sin       | DXIL::OpCode::Sin |   |
+| sinh      | DXIL::OpCode::Hsin |  |
+| tan       | DXIL::OpCode::Tan |   |
+| tanh      | DXIL::Opcode::Htan |  |
+
+## Math
+
+| Intrinsic | DXIL OPCode | Notes |
+|-----------|--------------|----------|
+| mod       | Emulated |              |
+| abs       | DXIL::OpCode::Imax |    |
+| ceil      | DXIL::OpCode::Round_pi ||
+| clamp     | DXIL::OpCode::UMax, UMin \ DXIL::OpCode::FMax/Fmin \ DXIL::OpCode::IMax/Imin ||
```

Comment:
This is what has me wondering if "emulated" means what I think it means. This is "emulated" as much as atan2 is. It may be useful to list the DXIL Ops it uses though.